pennylane/gradients/finite_difference.py

```diff
@@ -336,6 +337,10 @@ def finite_diff(
             validate_params=validate_params,
             shots=shots,
         )
+
+    if any(qml.math.get_dtype_name(p) == "float32" for p in tape.get_parameters()):
+        warn("Finite differences with float32 detected. Answers may be inaccurate. ", UserWarning)

```

Comment:
Nice! Please could we add a recommended action? E.g., it's not immediately clear what I'd need to do to stop the warning, such as to cast all my parameters to float64?

Reply:
added. now says to use float64.